platforms/ide/base-ide-plugins/src/main/java/org/gradle/plugins/ide/internal/IdePlugin.java

```diff
@@ -173,13 +174,8 @@ public void execute(Task task) {
         };
     }
 
-    protected static Action<Task> withGracefulDegradation(Project project) {
-        return task -> ((ProjectInternal) project).getServices()
-            .get(ConfigurationCacheDegradationController.class)
-            .requireConfigurationCacheDegradation(
-                task,
-                project.provider(() -> "Task is not compatible with the configuration cache")
-        );
+    protected static Action<Task> withGracefulDegradation() {
```

Comment:
Turns out this signature change breaks binary compatibilty... 😢 

If possible, we should try to deliver this to 9.0 RC2 instead, instead of introducing an incompatible change in the very next release.